### ISSUE TITLE
computed getter

### ISSUE THREAD

**Issue by hyusetiawan** [elbywan/hyperactiv#29]
Can we change the way computed works when it comes to getter functions? Currently, it's not listened to, though I can add a listener and set it as a property value instead, ex:
instead of `get value()` I create a computed then inside that callback I assign the new value to `.value`
Is it possible instead, to make the getter functions to be automatically computed? This behavior mimics mobx but i do see an improvement in DX as it is arguably  more straightforward

**Comment by elbywan** [comment]
Hey @hyusetiawan,

> instead of get value() I create a computed then inside that callback I assign the new value to .value

I'm not sure I understand your use case, as getters are supposed to retrieve values - not assign them.

Something like this:

```js
const { observe, computed } = hyperactiv

const observed = observe({
  _a: 1,
  get a() { return this._a },
  set a(v) { this._a = v },
  b: 2,
  c: 0
})

computed(() => {
  const { a, b } = observed
  console.log(`a + b = ${a + b}`)
})
// "a + b = 3"
observed.a = 2
// "a + b = 4"
observed.b = 3
// "a + b = 5"
```

Could you provide a code sample of what you would like to achieve?

**Comment by elbywan** [comment]
> Is it possible instead, to make the getter functions to be automatically computed?

Yes I guess it would be possible to add a new option to `observe` which would turn getter functions into computed.

I don't have a lot of spare time but I will look into it eventually 👍.
